### PR TITLE
[ADD]  Implement starknet_estimateMessageFee method

### LINKED ISSUE #80: Implement `starknet_estimateMessageFee` method
Implement `starknet_estimateMessageFee` rpc endpoint, according to the starknet-specs: 
* https://github.com/starkware-libs/starknet-specs/pull/100/files

Add method to the provider protocol, and provider implementation.

Also add tests for it. Note that if it's still not implemented in starknet-devnet when you start working on it, you might need to test it using a node. And best keep url of that node in an env variable.

### PR DESCRIPTION
**Pull request by dnKaratzas**
Resolves #80

### CODE REVIEW

**Review by bartekryba — CHANGES_REQUESTED** [review]

File: Sources/Starknet/Providers/StarknetProvider/JsonRpcParams.swift [hunk 1]
```diff
@@ -56,6 +56,18 @@ struct EstimateFeeParams: Encodable {
     }
 }
 
+struct EstimateMessageFeeParams: Encodable {
+    let message: StarknetCall
```
[the remark this answers is not in the document]
Comment (dnKaratzas) [reply Sources/Starknet/Providers/StarknetProvider/JsonRpcParams.swift:60]: We need to clarify as I tried that already and was getting the following error:

image

Please check https://github.com/starkware-libs/starknet-specs/commit/37c5ed1b5e4608cf8b80b3e06c726aadfa4f0db6
It is tagged for [v0.4.0-rc1](https://github.com/starkware-libs/starknet-specs/releases/tag/v0.4.0-rc1) but they seem to already using that schema

[the remark this answers is not in the document]
Comment (bartekryba) [reply Sources/Starknet/Providers/StarknetProvider/JsonRpcParams.swift:60]: I checked the commit, but it was actually a part of a PR, where it was changed to `MSG_FROM_L1` https://github.com/starkware-libs/starknet-specs/pull/100

[the remark this answers is not in the document]
Comment (dnKaratzas) [reply Sources/Starknet/Providers/StarknetProvider/JsonRpcParams.swift]: Fixed 6f9a005646676067dd295add03fdddfe0c39f9f7

**Review by bartekryba — APPROVED** [review]

**Comment by dnKaratzas** [comment]
Reverted back to 22785ab3d56b9d87f158664c90ad9b8d263d3808
To support `v0.3.1` schema

**Review by bartekryba — APPROVED** [review]

**Pull request merged**